src/shared/Core/EnvironmentBase.cs

```diff
@@ -124,7 +129,8 @@ public bool TryLocateExecutable(string program, out string path)
                 foreach (var basePath in paths)
                 {
                     string candidatePath = Path.Combine(basePath, program);
-                    if (FileSystem.FileExists(candidatePath))
+                    if (FileSystem.FileExists(candidatePath) &&
+                        !pathsToIgnore.Contains(candidatePath, StringComparer.OrdinalIgnoreCase))
```

Comment:
Update to also permit possible `null` arg

```suggestion
                    if (FileSystem.FileExists(candidatePath) && (pathsToIgnore is null ||
                        !pathsToIgnore.Contains(candidatePath, StringComparer.OrdinalIgnoreCase)))
```